src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -1,23 +1,53 @@
 package com.example.onlinebookstore.service;
 
+import com.example.onlinebookstore.dto.BookDto;
+import com.example.onlinebookstore.dto.CreateBookRequestDto;
+import com.example.onlinebookstore.exceptions.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.BookMapper;
 import com.example.onlinebookstore.model.Book;
 import com.example.onlinebookstore.repository.BookRepository;
 import java.util.List;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
-@RequiredArgsConstructor
 @Service
+@RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+        return bookMapper.toDto(bookRepository.save(bookMapper.toBook(createBookRequestDto)));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto getById(Long id) {
+        return bookRepository.findById(id)
+                .map(bookMapper::toDto)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Can't find book by this id: " + id));
+    }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto updateById(CreateBookRequestDto createBookRequestDto, Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't update book by this id: " + id));
+        BookDto bookDto = bookMapper.toDto(bookMapper.toBook(createBookRequestDto));
+        bookMapper.updateBookFromBookDto(bookDto, book);
```

Comment:
this is weird

you're performing double mapping for some reason ;)

Mapstruct method should work with `createDto`:
```suggestion
        bookMapper.updateBookFromBookDto(createBookRequestDto, book);
```